### PR TITLE
Added function to add sea 2km by 2km sea floor at a specified depth

### PR DESCRIPTION
**Pull request by jorisquist**
Added the `add_ocean_floor(depth)` function to the simulator so that it's easy to add a seafloor plane from LOB-star

### CODE REVIEW

File: lobster_simulator/simulator.py [hunk 1]
```diff
@@ -58,6 +59,13 @@ def __init__(self, time_step: int, config=None, gui=True):
         self._camera_position = Vec3([0, 0, 0])
         self._robot: Optional[AUV] = None
 
+    def add_ocean_floor(self, depth=100):
+        id = PybulletAPI.loadURDF(resource_filename("lobster_simulator", "data/plane1000.urdf"),
```
Comment (gijskoning) [inline lobster_simulator/simulator.py:63]: Could you add a small test for this function? At least that it covers accessing the urdf files. Preferably testing that it is actually creating an object in Pybullet

File: lobster_simulator/common/pybullet_api.py [hunk 2]
```diff
@@ -210,6 +210,14 @@ def createVisualCylinder(radius: float, height: float, color: List[float], orien
         shape = p.createVisualShape(p.GEOM_CYLINDER, radius=radius, length=height, rgbaColor=color)
         return p.createMultiBody(0, -1, shape, basePosition=[0, 0, 0], baseOrientation=orientation.asENU())
 
+    @staticmethod
+    def loadTexture(file_name: str) -> int:
+        return p.loadTexture(file_name)
+
+    @staticmethod
+    def changeVisualShape(object_id: int, textureUniqueId, rgbaColor):
```
Comment (gijskoning) [inline lobster_simulator/common/pybullet_api.py:218]: Can you add the types here? Might be good to specify that the rgbaColor needs to be an array with length of 4

File: test/unit_tests/test_simulator.py [hunk 3]
```diff
@@ -16,11 +16,18 @@ def test_add_ocean_floor(self):
 
         simulator = Simulator(4000, gui=False)
 
+        temp_load = PybulletAPI.loadURDF
+        temp_change_visual_shape = PybulletAPI.changeVisualShape
+
         PybulletAPI.loadURDF = Mock()
         PybulletAPI.changeVisualShape = Mock()
 
         simulator.add_ocean_floor(100)
```
Comment (gijskoning) [inline test/unit_tests/test_simulator.py]: Yes there is a better way. Use Mock.Patch() 
Example usage here: https://github.com/LOBSTER-Robotics/lobster-auv/blob/6de1edad5c42c8b7f2188dbde82705a84b518d27/test/integration/simulation_tests/test_depth_integrated.py
```suggestion
        with mock.patch("lobster_simulator.common.pybullet_api.PybulletAPI.loadURDF") as loadURDFMock:
            with mock.patch("lobster_simulator.common.pybullet_api.PybulletAPI.changeVisualShape") as changeVisualShapeMock:
                simulator.add_ocean_floor(100)


```

Comment (gijskoning) [reply test/unit_tests/test_simulator.py]: After applying suggestion, line 32 and 33 can be removed. Also lines 28-29 have to be incremented

Comment (gijskoning) [reply test/unit_tests/test_simulator.py]: You could also replace the 'with' lines with a decorator

Comment (jorisquist) [reply test/unit_tests/test_simulator.py]: Ah yes that is indeed a better solution

Comment (jorisquist) [reply test/unit_tests/test_simulator.py]: I have updated it

**Review by gijskoning — APPROVED** [review]

**Pull request merged**